app/drivers/sensor/trackball_pim447/trackball_pim447.c

```diff
@@ -0,0 +1,167 @@
+    int status;
+
+    status = trackball_pim447_read_reg(dev, reg_negative, &value_negative);
+    if (status < 0) {
+        return status;
+    }
+
+    status = trackball_pim447_read_reg(dev, reg_positive, &value_positive);
+    if (status < 0) {
+        return status;
+    }
+
+    *value = value_positive - value_negative;
+
+    return 0;
+}
+
+static int trackball_pim447_sample_fetch(const struct device *dev,
+                                         enum sensor_channel chan)
+{
+    struct trackball_pim447_data *data = dev->data;
+    int status;
+
+    if (chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_POS_DX) {
+        status = trackball_pim447_read_axis(dev,
+                                            TRACKBALL_PIM447_REG_LEFT,
+                                            TRACKBALL_PIM447_REG_RIGHT,
+                                            &data->dx);
+        if (status < 0) {
+            return status;
+        }
+    }
+
+    if (chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_POS_DY) {
+        status = trackball_pim447_read_axis(dev,
+                                            TRACKBALL_PIM447_REG_UP,
+                                            TRACKBALL_PIM447_REG_DOWN,
+                                            &data->dy);
+        if (status < 0) {
+            return status;
+        }
+    }
+
+    if (chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_POS_DZ) {
+        uint8_t value;
+        status = trackball_pim447_read_reg(dev,
+                                           TRACKBALL_PIM447_REG_SWITCH,
+                                           &value);
+        if (status < 0) {
+            return status;
+        }
+
+        data->dz = value;
+    }
+
+    return 0;
+}
+
+static int trackball_pim447_channel_get(const struct device *dev,
+                                        enum sensor_channel chan,
+                                        struct sensor_value *val)
+{
+    struct trackball_pim447_data *data = dev->data;
+
+    /* Not used.  */
+    val->val2 = 0;
+
+    switch (chan) {
+    case SENSOR_CHAN_POS_DX:
+        val->val1 = data->dx;
+        break;
+
+    case SENSOR_CHAN_POS_DY:
+        val->val1 = data->dy;
+        break;
+
+    case SENSOR_CHAN_POS_DZ:
+        val->val1 = data->dz;
+        break;
+
+    default:
+        return -ENOTSUP;
+    }
+
+    return 0;
+}
+
+static int trackball_pim447_init(const struct device *dev)
+{
+    struct trackball_pim447_data *data = dev->data;
+
+    data->i2c_dev = device_get_binding(DT_INST_BUS_LABEL(0));
+    if (data->i2c_dev == NULL) {
+        LOG_ERR("Failed to get I2C device");
+        return -EINVAL;
+    }
+
+    return 0;
+}
+
+static struct trackball_pim447_data trackball_pim447_data;
+
+static const struct sensor_driver_api trackball_pim447_api = {
+    .sample_fetch = trackball_pim447_sample_fetch,
+    .channel_get  = trackball_pim447_channel_get,
+};
+
+DEVICE_DT_INST_DEFINE(0, &trackball_pim447_init, device_pm_control_nop,
```

Comment:
```suggestion
DEVICE_DT_INST_DEFINE(0, &trackball_pim447_init, NULL,
```

`device_pm_control_nop` has been deprecated in favor of NULL, per [these Zephy release notes](https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/releases/release-notes-2.6.rst).